src/esperanza/finalizationstate.cpp

```diff
@@ -629,21 +629,8 @@ Result FinalizationState::CalculateWithdrawAmount(const uint160 &validatorAddres
                 validatorAddress.GetHex());
   }
 
-  const auto &validator = it->second;
-
-  uint32_t endDynasty = validator.m_end_dynasty;
-
-  if (m_current_dynasty <= endDynasty) {
-    return fail(Result::WITHDRAW_BEFORE_END_DYNASTY,
-                /*log_errors=*/true,
-                "%s: Too early to withdraw, minimum expected dynasty for "
-                "withdraw is %d.\n",
-                __func__, endDynasty);
-  }
```

Comment:
It's actually correct, once you reach this epoch, you'll able to withdraw without the logout but it will take you ~55 233 years. I suggest once 2/3 of this time elapsed we open a brainstorm issue and link it in TODO how we will migrating to `uint64_t` as it can cause a severe issue to the protocol. I considered adding a TODO in this PR but as we prefer to link GitHub issue too, I decided to postpone it for now.

Reply:
I'm not sure it's correct, really. After ~55 233 years it will overflow somewhere :)

I suggest getting meaningful description back.

Reply:
This is a meaningful description. I removed `WITHDRAW_BEFORE_END_DYNASTY` on purpose because it was shadowed right after the call of this function, was not used anywhere and didn't impact how node behaves/DoS peers.

Removing it we have 1 less state to support and synchronize. Now the correct way to check in which state the finalizer is is to ask `FinalizationState` and not relying/decoding the error code.

Reply:
Ok, I misread this piece of change a bit. There was 4294967295 in the log anyway.